sonic-xcvrd/xcvrd/xcvrd_utilities/xcvrd_common.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python3
+
+"""
+    xcvrd_common
+    Common utilities for xcvrd daemon components
+"""
+
+try:
+    import sys
+    import subprocess
+    import traceback
+    from swsscommon import swsscommon
+    from sonic_py_common import syslogger
+    from . import sfp_status_helper
+    from sonic_platform_base.sonic_xcvr.api.public.c_cmis import CmisApi
+
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+SYSLOG_IDENTIFIER = "xcvrd_common"
+
+# Global variables that will be injected from the parent module
+platform_chassis = None
+platform_sfputil = None
+helper_logger = syslogger.SysLogger(SYSLOG_IDENTIFIER, enable_runtime_config=True)
+
+def init_globals(chassis, sfputil, logger):
+    """Initialize global variables with injected dependencies"""
+    global platform_chassis, platform_sfputil, helper_logger
+    platform_chassis = chassis
+    platform_sfputil = sfputil
+    helper_logger = logger
+
+def log_exception_traceback():
+    """Log exception traceback using the helper logger"""
+    exc_type, exc_value, exc_traceback = sys.exc_info()
+    msg = traceback.format_exception(exc_type, exc_value, exc_traceback)
+    for tb_line in msg:
+        for tb_line_split in tb_line.splitlines():
+            helper_logger.log_error(tb_line_split)
+
+def update_port_transceiver_status_table_sw(logical_port_name, status_sw_tbl, status, error_descriptions='N/A'):
+    """Update port SFP status table for SW fields on receiving SFP change event"""
+    fvs = swsscommon.FieldValuePairs([('status', status), ('error', error_descriptions)])
+    status_sw_tbl.set(logical_port_name, fvs)
+
+def _wrapper_get_presence(physical_port):
```

Comment:
@bobby-nexthop there are many _wrapper in xcvrd.py. would it possible to clean all of them and just use actual function name. like

`_wrapper_get_presence()` -> `get_presence()`. Around the year 2021 when different platforms were transitioning to platform API 2.0, we provided this interim solution to use a wrapper function to use either platform API or the sfputil API, but by now all platforms should be on platform APIs.

Use platform chassis API and no need to fallback on sfputil.